x-pack/plugins/security_solution/public/common/components/threat_match/helpers.tsx

```diff
@@ -90,10 +95,11 @@ export const getEntryOnFieldChange = (
   const { entryIndex } = item;
   return {
     updatedEntry: {
+      id: item.id,
       field: newField != null ? newField.name : '',
       type: 'mapping',
       value: item.value != null ? item.value.name : '',
-    },
+    } as Entry, // Cast to Entry on purpose to be able to add the id for ReactJS rendering purposes.
```

Comment:
IMO this comment isn't entirely clear. Maybe: 
```suggestion
    } as Entry, // Cast to Entry since id is only used as a react key prop and can be ignored elsewhere
```